src/embed/connection.rs

```diff
@@ -82,7 +81,7 @@ async fn connection_future<'a, 'b>(
     UnorderedResult::None()
 }
 
-pub async fn process(store: dag::Store, rx: Receiver<Request>) {
+pub async fn process(store: dag::Store, rx: Receiver<Request>, client_id: String) {
```

Comment:
I think I need to reread the `String` vs `&str` part but are we not copying the string too much here?

Reply:
`String` is a move type so what's happening here is it's moving into `process`, not getting copied. You can verify this by looking for the `Copy` trait here:

https://doc.rust-lang.org/std/string/struct.String.html#trait-implementations

By default things in Rust are move unless `Copy` is implemented.

Reply:
(the places you see `.clone()` we are explicitly copying the string, this is sort of unavoidable given current shape of things.